spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/MessageConverterConfigurer.java

```diff
@@ -185,116 +193,79 @@ public int selectPartition(Object key, int partitionCount) {
 
 	}
 
-	private final class ContentTypeConvertingInterceptor extends ChannelInterceptorAdapter {
-
-		private final String contentType;
+	private final class ContentTypeConvertingInterceptor
+			extends ChannelInterceptorAdapter {
 
 		private final MimeType mimeType;
 
 		private final boolean input;
 
-		private final Class<?> klazz;
-
 		private final MessageConverter messageConverter;
 
 		private final boolean provideHint;
 
 		private ContentTypeConvertingInterceptor(String contentType, boolean input) {
-			this.contentType = contentType;
 			this.mimeType = MessageConverterUtils.getMimeType(contentType);
 			this.input = input;
-			if (MessageConverterUtils.X_JAVA_OBJECT.includes(this.mimeType)) {
-				this.klazz = MessageConverterUtils
-						.getJavaTypeForJavaObjectContentType(this.mimeType);
-			}
-			else if (this.mimeType.equals(MessageConverterUtils.X_SPRING_TUPLE)) {
-				this.klazz = Tuple.class;
-			}
-			else if (this.mimeType.getType().equals("text") || this.mimeType.getSubtype().equals(
-					"json") || this.mimeType.getSubtype().equals("xml")) {
-				this.klazz = String.class;
-			}
-			else {
-				this.klazz = byte[].class;
-			}
+
 			this.messageConverter = MessageConverterConfigurer.this.compositeMessageConverterFactory
-					.getMessageConverterForType(this.mimeType);
+					.getMessageConverterForAllRegistered();
 			this.provideHint = this.messageConverter instanceof AbstractMessageConverter;
 		}
 
 		@Override
 		public Message<?> preSend(Message<?> message, MessageChannel channel) {
```

Comment:
We don't seem to be using `provideHint` during the conversion. Is there a specific reason for that?

Reply:
Yes, to be honest I never understood why we needed in first place. On the way in, the conversion is actually handled by MethodInvokerHandlers, and they will provide the target Class<?> every time.

The whole klazz logic is now gone.